code/game/objects/items/stacks/sheets/mineral.dm

```diff
@@ -218,10 +218,20 @@ GLOBAL_LIST_INIT(sandbag_recipes, list ( \
 
 /obj/item/stack/sheet/mineral/plasma/welder_act(mob/user, obj/item/I)
```

Comment:
This one is missing a `return TRUE` at the end. Else the system does not know if the welder action succeeded or not

Reply:
I thought if(I.use_tool(src, user, volume = I.tool_volume)) made sure it would only do it if the welder had fuel or not

Reply:
It will yes.
Returning true will tell the system to not do an attackby since the "attack" is already handled in welder_act. Without it the system thinks that nothing happened in welder_act and thus that it still needs to do an attackby